File: app/src/main/java/br/com/vicentec12/mygames/data/source/Local.java
package br.com.vicentec12.mygames.data.source;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Qualifier;

/**
 * Qualifier criado para diferenciar para o Dagger quando for preciso de uma instância
 * de um DataSource ele entregar um LocalDataSource.
 */
@Documented
@Qualifier
@Retention(RetentionPolicy.RUNTIME)
public @interface Local {
}

File: app/src/main/java/br/com/vicentec12/mygames/data/source/Local.kt
package br.com.vicentec12.mygames.data.source

import javax.inject.Qualifier

/**
 * Qualifier criado para diferenciar para o Dagger quando for preciso de uma instância
 * de um DataSource ele entregar um LocalDataSource.
 */
@MustBeDocumented
@Qualifier
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class Local